### DISCUSSION TITLE
Versioning on CDK Project Types

### DISCUSSION THREAD

**Discussion by automartin5000** [projen/projen#3348] [Q&A]
I just noticed that (on a quick glance) neither the Typescript nor Python CDK project types do any kind of version bumping/git tagging. Is that deliberate? The Python project expects an explicitly declared version arg (which the doc seemingly incorrectly refers to as Python version), but the Typescript one doesn't.

Just wondering if there was an intentional decision here, or if it's just an incomplete implementation. Either way, would be interested in people's thoughts here.

**Comment by mrgrain** [comment]
Could you elaborate a bit more? I don't I understand what you are referring to. Code examples or links to docs always help.

**Comment by automartin5000** [comment]
@mrgrain Sure. Take the awscdk-app-py project type. Here's an example [tasks.json](https://github.com/projen/projen/files/14229563/tasks.json) it creates. There doesn't appear to be anywhere where it runs `bump` and that task doesn't exist. Also, at least for Python projects, there seems to be a [version](https://github.com/projen/projen/blob/4f3ab75ced2525aca939424e28db62dbd7c0fa5e/src/python/python-project.ts#L208) option that sets a version in `pyproject.toml`. So that seems to conflict with the idea of [keeping versions at 0.0.0](https://github.com/projen/projen/issues/1141#issuecomment-948984418) and using git tags to maintain versions.

**Reply by mrgrain** [reply]
Thanks for the explanation, this helps!

**Comment by mrgrain** [comment]
Projen has a loosely defined concept of "libraries" which are reusable, published packages and "apps", which are deployed (publicly or privately), but not released. For AWS CDK apps this is probably the most well defined. Because CDK Construct libraries need to be build in a certain way. 

Either way, the idea isn't really to prevent anyone from doing anything. It's more that some default setups might be different. Whatever your Node project looks like you can always use the `Release` component.


Regarding python, it's just that we don't have a ready to use python `Release` component yet. The one for node has been built fairly general purpose, but there are a few bits that currently don't translate/are missing for python.
